### PR TITLE
add(mc): new cloud-azure-database-postgresql MC

### PR DESCRIPTION
**Pull request by omercier**
## Description

New Monitoring Connector

## Target version (i.e. version that this PR changes)

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [ ] 24.04.x
- [ ] Cloud
- [x] Monitoring Connectors

### CODE REVIEW

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-azure-database-postgresql.md [hunk 1]
```diff
@@ -0,0 +1,638 @@
...
+| AZURERESOURCEGROUP | Set resource group (required if resource's name is used)                                                                   |                   | X           |
+| AZURESUBSCRIPTION  | Set Azure subscription (required if logged to several subscriptions)                                                       |                   | X           |
+| AZURETENANT        | Set Azure tenant ID                                                                                                        |                   | X           |
+| PROXYURL           | Proxy URL if any                                                                                                           |                   |             |
+| EXTRAOPTIONS       | Any extra option you may want to add to every command (a --verbose flag for example). Toutes les options sont listées [ici](#options-disponibles).                       |                   |             |
+
+> Paramétrez les options suivantes dans la macro EXTRAOPTIONS si vous superviser des ressources Microsoft Azure géré par 21Vianet (Azure China):
```
Comment (cg-tw) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-azure-database-postgresql.md:214]:
```suggestion
> Paramétrez les options suivantes dans la macro EXTRAOPTIONS si vous souhaitez superviser des ressources Microsoft Azure gérées par 21Vianet (Azure China):
```

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-azure-database-postgresql.md [hunk 2]
```diff
@@ -0,0 +1,638 @@
...
+| TIMEFRAME              | Set timeframe in seconds (e.g. '3600' to check last 60 minutes).                                   | 900               |             |
+| INTERVAL               | Set interval of the metric query (can be : PT1M, PT5M, PT15M, PT30M, PT1H, PT6H, PT12H, PT24H)                                                     | PT5M              |             |
+| AGGREGATION            | Define how the data must be aggregated. Available aggregations: 'minimum', 'maximum', 'average', 'total' and 'count'. Can be called multiple times | Maximum             |             |
+| FILTERMETRIC           | Filter metrics (available metrics: 'Backup Storage used', 'Server Log storage used', 'Storage Percent', 'Storage Used'). |                   |             |
+| FILTERDIMENSION        | Specify the metric dimension (required for some specific metrics) Syntax example: --filter-dimension="$metricname eq '$metricvalue'". |
+| WARNINGSERVERLOGUSAGE  | Thresholds where '*'                                                                               |                   |             |
```
Comment (cg-tw) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-azure-database-postgresql.md:326]: where what?

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-azure-database-postgresql.md [hunk 3]
```diff
@@ -0,0 +1,638 @@
...
+|:-------------------|:---------------------------------------------------------------------------------------------------|:------------------|:-----------:|
+| TIMEFRAME          | Set timeframe in seconds (e.g. '3600' to check last 60 minutes).                                   | 900               |             |
+| INTERVAL           | Set interval of the metric query (can be : PT1M, PT5M, PT15M, PT30M, PT1H, PT6H, PT12H, PT24H)                                                     | PT5M              |             |
+| AGGREGATION        | Define how the data must be aggregated. Available aggregations: 'minimum', 'maximum', 'average', 'total' and 'count'. Can be called multiple times | Average             |             |
+| FILTERMETRIC       | Filter metrics (available metrics: 'Network Out', 'Network In'). |                   |             |
+| FILTERDIMENSION    | Specify the metric dimension (required for some specific metrics) Syntax example: --filter-dimension="$metricname eq '$metricvalue'". |
+| WARNINGTRAFFICIN   | Thresholds where '*'                                                                               |                   |             |
```
Comment (cg-tw) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-azure-database-postgresql.md:346]: where what?

File: pp/integrations/plugin-packs/procedures/cloud-azure-database-postgresql.md [hunk 4]
```diff
@@ -0,0 +1,637 @@
...
+* **Cloud-Azure-Database-PostgreSQL-custom**
+
+The connector brings the following service templates (sorted by the host template they are attached to):
+
+<Tabs groupId="sync">
+<TabItem value="Cloud-Azure-Database-PostgreSQL-custom" label="Cloud-Azure-Database-PostgreSQL-custom">
+
+| Service Alias  | Service Template                                          | Service Description                                          |
+|:---------------|:----------------------------------------------------------|:-------------------------------------------------------------|
+| Connections    | Cloud-Azure-Database-PostgreSQL-Connections-Api-custom    | Monitor Azure Database for PostgreSQL connections statistics |
```
Comment (cg-tw) [inline pp/integrations/plugin-packs/procedures/cloud-azure-database-postgresql.md:23]:
```suggestion
| Connections    | Cloud-Azure-Database-PostgreSQL-Connections-Api-custom    | Monitor Azure Database for PostgreSQL connection statistics |
```

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-azure-database-postgresql.md [hunk 5]
```diff
@@ -0,0 +1,638 @@
...
+
+Le connecteur apporte les modèles de service suivants
+(classés selon le modèle d'hôte auquel ils sont rattachés) :
+
+<Tabs groupId="sync">
+<TabItem value="Cloud-Azure-Database-PostgreSQL-custom" label="Cloud-Azure-Database-PostgreSQL-custom">
+
+| Alias          | Modèle de service                                         | Description                                                                         |
+|:---------------|:----------------------------------------------------------|:------------------------------------------------------------------------------------|
+| Connections    | Cloud-Azure-Database-PostgreSQL-Connections-Api-custom    | Contrôle les statistiques de connexions de l'instance Azure Database for PostgreSQL |
```
Comment (cg-tw) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-azure-database-postgresql.md:24]:
```suggestion
| Connections    | Cloud-Azure-Database-PostgreSQL-Connections-Api-custom    | Contrôle les statistiques de connexion de l'instance Azure Database for PostgreSQL |
```

File: pp/integrations/plugin-packs/procedures/cloud-azure-database-postgresql.md [hunk 6]
```diff
@@ -0,0 +1,637 @@
...
+| AZURERESOURCEGROUP | Set resource group (required if resource's name is used)                                                                   |                   | X           |
+| AZURESUBSCRIPTION  | Set Azure subscription (required if logged to several subscriptions)                                                       |                   | X           |
+| AZURETENANT        | Set Azure tenant ID                                                                                                        |                   | X           |
+| PROXYURL           | Proxy URL if any                                                                                                           |                   |             |
+| EXTRAOPTIONS       | Any extra option you may want to add to every command (a --verbose flag for example). All options are listed [here](#available-options).                       |                   |             |
+
+> Set the following options in the EXTRAOPTIONS macro if you are monitoring resource from Microsoft Azure operated by 21Vianet (Azure China):
```
Comment (cg-tw) [inline pp/integrations/plugin-packs/procedures/cloud-azure-database-postgresql.md:215]:
```suggestion
> Set the following options in the EXTRAOPTIONS macro if you are monitoring resources from Microsoft Azure operated by 21Vianet (Azure China):
```

File: pp/integrations/plugin-packs/procedures/cloud-azure-database-postgresql.md [hunk 7]
```diff
@@ -0,0 +1,637 @@
...
+| TIMEFRAME              | Set timeframe in seconds (e.g. '3600' to check last 60 minutes).                                   | 900               |             |
+| INTERVAL               | Set interval of the metric query (can be : PT1M, PT5M, PT15M, PT30M, PT1H, PT6H, PT12H, PT24H)                                                     | PT5M              |             |
+| AGGREGATION            | Define how the data must be aggregated. Available aggregations: 'minimum', 'maximum', 'average', 'total' and 'count'. Can be called multiple times | Maximum             |             |
+| FILTERMETRIC           | Filter metrics (available metrics: 'Backup Storage used', 'Server Log storage used', 'Storage Percent', 'Storage Used'). |                   |             |
+| FILTERDIMENSION        | Specify the metric dimension (required for some specific metrics) Syntax example: --filter-dimension="$metricname eq '$metricvalue'". |
+| WARNINGSERVERLOGUSAGE  | Thresholds where '*'                                                                               |                   |             |
```
Comment (cg-tw) [inline pp/integrations/plugin-packs/procedures/cloud-azure-database-postgresql.md:327]: where what?

File: pp/integrations/plugin-packs/procedures/cloud-azure-database-postgresql.md [hunk 8]
```diff
@@ -0,0 +1,637 @@
...
+|:-------------------|:---------------------------------------------------------------------------------------------------|:------------------|:-----------:|
+| TIMEFRAME          | Set timeframe in seconds (e.g. '3600' to check last 60 minutes).                                   | 900               |             |
+| INTERVAL           | Set interval of the metric query (can be : PT1M, PT5M, PT15M, PT30M, PT1H, PT6H, PT12H, PT24H)                                                     | PT5M              |             |
+| AGGREGATION        | Define how the data must be aggregated. Available aggregations: 'minimum', 'maximum', 'average', 'total' and 'count'. Can be called multiple times | Average             |             |
+| FILTERMETRIC       | Filter metrics (available metrics: 'Network Out', 'Network In'). |                   |             |
+| FILTERDIMENSION    | Specify the metric dimension (required for some specific metrics) Syntax example: --filter-dimension="$metricname eq '$metricvalue'". |
+| WARNINGTRAFFICIN   | Thresholds where '*'                                                                               |                   |             |
```
Comment (cg-tw) [inline pp/integrations/plugin-packs/procedures/cloud-azure-database-postgresql.md:347]: where what?

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-azure-database-postgresql.md [hunk 9]
```diff
@@ -0,0 +1,638 @@
...
+telle que celle-ci (remplacez les valeurs d'exemple par les vôtres) :
+
+```bash
+/usr/lib/centreon/plugins/centreon_azure_database_postgres_api.pl \
+	--plugin=cloud::azure::database::postgres::plugin \
+	--mode=traffic \
+	--custommode='api' \
+	--resource='' \
+	--resource-group='' \
+	--subscription='' \
```
Comment (cg-tw) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-azure-database-postgresql.md:372]: dummy values for mandatory parameters?

File: pp/integrations/plugin-packs/procedures/cloud-azure-database-postgresql.md [hunk 10]
```diff
@@ -0,0 +1,637 @@
...
+is able to monitor an Azure Instance using a command like this one (replace the sample values by yours):
+
+```bash
+/usr/lib/centreon/plugins/centreon_azure_database_postgres_api.pl \
+	--plugin=cloud::azure::database::postgres::plugin \
+	--mode=traffic \
+	--custommode='api' \
+	--resource='' \
+	--resource-group='' \
+	--subscription='' \
```
Comment (cg-tw) [inline pp/integrations/plugin-packs/procedures/cloud-azure-database-postgresql.md:371]: dummy values for mandatory parameters?

**Review by cg-tw — CHANGES_REQUESTED** [review]

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-azure-database-postgresql.md [hunk 11]
```diff
@@ -0,0 +1,638 @@
...
+| CRITICALSERVERLOGUSAGE | Thresholds where '*'                                                                               |                   |             |
+| WARNINGSTORAGEBACKUP   | Thresholds where '*'                                                                               |                   |             |
+| CRITICALSTORAGEBACKUP  | Thresholds where '*'                                                                               |                   |             |
+| WARNINGSTORAGEPERCENT  | Thresholds where '*'                                                                               |                   |             |
+| CRITICALSTORAGEPERCENT | Thresholds where '*'                                                                               |                   |             |
+| WARNINGSTORAGEUSED     | Thresholds where '*'                                                                               |                   |             |
+| CRITICALSTORAGEUSED    | Thresholds where '*'                                                                               |                   |             |
```
Comment (omercier) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-azure-database-postgresql.md:333]:
```suggestion
| WARNINGSERVERLOGUSAGE  | Thresholds                                                                               |                   |             |
| CRITICALSERVERLOGUSAGE | Thresholds                                                                               |                   |             |
| WARNINGSTORAGEBACKUP   | Thresholds                                                                              |                   |             |
| CRITICALSTORAGEBACKUP  | Thresholds                                                                               |                   |             |
| WARNINGSTORAGEPERCENT  | Thresholds                                                                               |                   |             |
| CRITICALSTORAGEPERCENT | Thresholds                                                                               |                   |             |
| WARNINGSTORAGEUSED     | Thresholds                                                                                |                   |             |
| CRITICALSTORAGEUSED    | Thresholds                                                                                |                   |             |
```

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-azure-database-postgresql.md [hunk 12]
```diff
@@ -0,0 +1,638 @@
...
+| AGGREGATION        | Define how the data must be aggregated. Available aggregations: 'minimum', 'maximum', 'average', 'total' and 'count'. Can be called multiple times | Average             |             |
+| FILTERMETRIC       | Filter metrics (available metrics: 'Network Out', 'Network In'). |                   |             |
+| FILTERDIMENSION    | Specify the metric dimension (required for some specific metrics) Syntax example: --filter-dimension="$metricname eq '$metricvalue'". |
+| WARNINGTRAFFICIN   | Thresholds where '*'                                                                               |                   |             |
+| CRITICALTRAFFICIN  | Thresholds where '*'                                                                               |                   |             |
+| WARNINGTRAFFICOUT  | Thresholds where '*'                                                                               |                   |             |
+| CRITICALTRAFFICOUT | Thresholds where '*'                                                                               |                   |             |
```
Comment (omercier) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-azure-database-postgresql.md:349]:
```suggestion
| WARNINGTRAFFICIN   | Thresholds                                                                                 |                   |             |
| CRITICALTRAFFICIN  | Thresholds                                                                                 |                   |             |
| WARNINGTRAFFICOUT  | Thresholds                                                                                |                   |             |
| CRITICALTRAFFICOUT | Thresholds                                                                                |                   |             |
```

File: pp/integrations/plugin-packs/procedures/cloud-azure-database-postgresql.md (hunk 4 above)
Comment (omercier) [reply pp/integrations/plugin-packs/procedures/cloud-azure-database-postgresql.md:23]: On ne teste pas la connexion, mais on donne des stats sur le nombre de connexions

File: pp/integrations/plugin-packs/procedures/cloud-azure-database-postgresql.md [hunk 13]
```diff
@@ -0,0 +1,637 @@
...
+| CRITICALSERVERLOGUSAGE | Thresholds where '*'                                                                               |                   |             |
+| WARNINGSTORAGEBACKUP   | Thresholds where '*'                                                                               |                   |             |
+| CRITICALSTORAGEBACKUP  | Thresholds where '*'                                                                               |                   |             |
+| WARNINGSTORAGEPERCENT  | Thresholds where '*'                                                                               |                   |             |
+| CRITICALSTORAGEPERCENT | Thresholds where '*'                                                                               |                   |             |
+| WARNINGSTORAGEUSED     | Thresholds where '*'                                                                               |                   |             |
+| CRITICALSTORAGEUSED    | Thresholds where '*'                                                                               |                   |             |
```
Comment (omercier) [inline pp/integrations/plugin-packs/procedures/cloud-azure-database-postgresql.md:334]:
```suggestion
| WARNINGSERVERLOGUSAGE  | Thresholds                                                                                  |                   |             |
| CRITICALSERVERLOGUSAGE | Thresholds                                                                                 |                   |             |
| WARNINGSTORAGEBACKUP   | Thresholds                                                                                 |                   |             |
| CRITICALSTORAGEBACKUP  | Thresholds                                                                                 |                   |             |
| WARNINGSTORAGEPERCENT  | Thresholds                                                                                |                   |             |
| CRITICALSTORAGEPERCENT | Thresholds                                                                                 |                   |             |
| WARNINGSTORAGEUSED     | Thresholds                                                                                   |                   |             |
| CRITICALSTORAGEUSED    | Thresholds                                                                                 |                   |             |
```

File: pp/integrations/plugin-packs/procedures/cloud-azure-database-postgresql.md [hunk 14]
```diff
@@ -0,0 +1,637 @@
...
+| AGGREGATION        | Define how the data must be aggregated. Available aggregations: 'minimum', 'maximum', 'average', 'total' and 'count'. Can be called multiple times | Average             |             |
+| FILTERMETRIC       | Filter metrics (available metrics: 'Network Out', 'Network In'). |                   |             |
+| FILTERDIMENSION    | Specify the metric dimension (required for some specific metrics) Syntax example: --filter-dimension="$metricname eq '$metricvalue'". |
+| WARNINGTRAFFICIN   | Thresholds where '*'                                                                               |                   |             |
+| CRITICALTRAFFICIN  | Thresholds where '*'                                                                               |                   |             |
+| WARNINGTRAFFICOUT  | Thresholds where '*'                                                                               |                   |             |
+| CRITICALTRAFFICOUT | Thresholds where '*'                                                                               |                   |             |
```
Comment (omercier) [inline pp/integrations/plugin-packs/procedures/cloud-azure-database-postgresql.md:350]:
```suggestion
| WARNINGTRAFFICIN   | Thresholds                                                                                |                   |             |
| CRITICALTRAFFICIN  | Thresholds                                                                                 |                   |             |
| WARNINGTRAFFICOUT  | Thresholds                                                                                |                   |             |
| CRITICALTRAFFICOUT | Thresholds                                                                                |                   |             |
```

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-azure-database-postgresql.md [hunk 15]
```diff
@@ -0,0 +1,638 @@
...
+/usr/lib/centreon/plugins/centreon_azure_database_postgres_api.pl \
+	--plugin=cloud::azure::database::postgres::plugin \
+	--mode=traffic \
+	--custommode='api' \
+	--resource='' \
+	--resource-group='' \
+	--subscription='' \
+	--tenant='' \
+	--client-id='' \
+	--client-secret='' \
```
Comment (omercier) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-azure-database-postgresql.md:375]:
```suggestion
	--subscription='xxxxxxxxxxxxxx' \
	--tenant='xxxxxxxxxxxxxx' \
	--client-id='xxxxxxxxxxxxxx' \
	--client-secret='xxxxxxxxxxxxxx' \
```

File: pp/integrations/plugin-packs/procedures/cloud-azure-database-postgresql.md [hunk 16]
```diff
@@ -0,0 +1,637 @@
...
+/usr/lib/centreon/plugins/centreon_azure_database_postgres_api.pl \
+	--plugin=cloud::azure::database::postgres::plugin \
+	--mode=traffic \
+	--custommode='api' \
+	--resource='' \
+	--resource-group='' \
+	--subscription='' \
+	--tenant='' \
+	--client-id='' \
+	--client-secret='' \
```
Comment (omercier) [inline pp/integrations/plugin-packs/procedures/cloud-azure-database-postgresql.md:374]:
```suggestion
	--subscription='xxxxxxxxxxxxxx' \
	--tenant='xxxxxxxxxxxxxx' \
	--client-id='xxxxxxxxxxxxxx' \
	--client-secret='xxxxxxxxxxxxxx' \
```

File: pp/integrations/plugin-packs/procedures/cloud-azure-database-postgresql.md (hunk 4 above)
Comment (cg-tw) [inline pp/integrations/plugin-packs/procedures/cloud-azure-database-postgresql.md:23]:
```suggestion
| Connections    | Cloud-Azure-Database-PostgreSQL-Connections-Api-custom    | Monitor the number of connections to the Azure PostgreSQL Database |
```

**Review by cg-tw — CHANGES_REQUESTED** [review]

**Review by cg-tw — APPROVED** [review]

**Pull request merged**